bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/device/Device.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hapero.internal.device;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Device} class contains the raw
+ * data received from the heating control system and
+ * conversion functions.
+ *
+ * @author Daniel Walter - Initial contribution
+ */
+@NonNullByDefault
+public class Device {
+    private final Logger logger = LoggerFactory.getLogger(Device.class);
+    /** Raw data stream from the device */
+    protected String[] dataItems = {};
+
+    /**
+     * Constructor
+     *
+     * @param items List of data items as Strings
+     */
+    public Device(String[] items) {
+        dataItems = items;
+    }
+
+    /**
+     * @param dataItems
+     */
+    public void setDataItems(String[] dataItems) {
+        this.dataItems = dataItems;
+    }
+
+    /**
+     * Returns the value of the dataItem at Index as a Float
+     *
+     * @param index Index of the dataItem to return
+     * @return Float or null if the index is invalid or the dataItem can not be converted to Float
+     */
+    public @Nullable Float getFloat(int index) {
+        Float result = null;
+
+        if (index >= 0 && index < dataItems.length) {
+            try {
+                result = Float.valueOf(dataItems[index]);
+            } catch (NumberFormatException e) {
+                logger.warn("getFloat: {}.", e.getMessage());
```

Comment:
Consider throwing a dedicated exception here and log it with more context in the handler. The reason would be provided with the exception, and the handler would know which channel could not be updated because of this unexpected value or bug.

This applies to all warnings logged by this class.